src/python/pants/engine/target_test.py

```diff
@@ -498,6 +499,27 @@ def nested():
     assert nested() == nested()
 
 
+def test_coarsened_target_closure() -> None:
+    all_targets = [FortranTarget({}, Address(name)) for name in string.ascii_lowercase[:5]]
+    a, b, c, d, e = all_targets
+
+    def ct(members: List[Target], dependencies: List[CoarsenedTarget] = []) -> CoarsenedTarget:
+        return CoarsenedTarget(members, dependencies)
+
+    def assert_closure(cts: Sequence[CoarsenedTarget], expected: Sequence[Target]) -> None:
+        assert sorted(t.address for t in CoarsenedTargets(cts).closure()) == sorted(
+            t.address for t in expected
+        )
+
+    ct1 = ct([a], [])
+    ct2 = ct([b, c], [ct1])
+    ct3 = ct([d, e], [ct1, ct2])
```

Comment:
Haven't fully grokked CTs yet: should we test for overlapping `members` between CTs as well?

Reply:
No: that should be impossible due to how `CTs` are constructed. And that bit has existing tests: https://github.com/pantsbuild/pants/blob/83b5f04cee786e6e0a0f1ad8dd3a80fb45c7fcde/src/python/pants/engine/internals/graph_test.py#L370-L449